tools/functional-tester/etcd-agent/agent.go

```diff
@@ -102,9 +102,11 @@ func (a *Agent) restart() error {
 }
 
 func (a *Agent) cleanup() error {
-	err := a.stop()
-	if err != nil {
-		return err
+	if a.state == stateStarted {
```

Comment:
when we call stop, we always want it to be stopped. we do not care if it was stopped or not.